lib/experimental/terms-query-block.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Terms Query block helpers.
+ *
+ * @since 6.9.0
+ * @package gutenberg
+ * @subpackage Blocks
+ */
+
+if ( ! function_exists( 'build_query_vars_from_terms_query_block' ) ) {
+	/**
+	 * Helper function that constructs a WP_Term_Query args array from a Terms Query
+	 * block's attributes.
+	 *
+	 * @param WP_Block $block Block instance.
+	 * @param int      $page Current query's page.
+	 *
+	 * @return array Returns the constructed WP_Term_Query arguments.
+	 */
+	function build_query_vars_from_terms_query_block( $block, $page = 1 ) {
+		$query = array(
+			'taxonomy'     => 'category',
+			'number'       => 10,
+			'order'        => 'asc',
+			'orderby'      => 'name',
+			'hide_empty'   => true,
+			'parent'       => 0,
+			'hierarchical' => false,
+		);
+
+		if ( isset( $block->context['termQuery'] ) ) {
+			if ( ! empty( $block->context['termQuery']['taxonomy'] ) ) {
+				$taxonomy_param = $block->context['termQuery']['taxonomy'];
+				if ( is_taxonomy_viewable( $taxonomy_param ) ) {
+					$query['taxonomy'] = $taxonomy_param;
+				}
+			}
+			if ( isset( $block->context['termQuery']['exclude'] ) ) {
+				$excluded_term_ids = array_map( 'intval', $block->context['termQuery']['exclude'] );
+				$excluded_term_ids = array_filter( $excluded_term_ids );
+				$query['exclude']  = $excluded_term_ids;
```

Comment:
Is there any control or even a path code that uses `exclude`?

Reply:
I guess the same applies for `include`?

Reply:
And probably for `perPage` and `page`? Sorry for asking the questions here but trying to understand what already exists and what's the vision for this block. I'll also ping @mikachan because my questions are about the original implementation too.

We might have included too many stuff similar to Query block that might not be needed from the start and it would be better to try to simplify as much as we can and even avoid some mistakes (not something specific right now in my mind) that we did with Query.

`page` in Query is used for pagination and I think terms query doesn't have pagination. I don't think it's used in code right now.. 

`perPage` seems to be the max-terms which is not named properly and not even used in `query`, so it should be a top level attribute.

Reply:
I open a [PR](https://github.com/WordPress/gutenberg/pull/71916) to remove some redundant attributes. Mentioning here as I couldn't add you as reviewer explicitly.